utilities/transactions/write_prepared_txn.cc

```diff
@@ -232,19 +221,31 @@ Status WritePreparedTxn::RollbackInternal() {
   }
   // The Rollback marker will be used as a batch separator
   WriteBatchInternal::MarkRollback(&rollback_batch, name_);
-  const bool disable_memtable = true;
+  const bool DISABLE_MEMTABLE = true;
   const uint64_t no_log_ref = 0;
   uint64_t seq_used = kMaxSequenceNumber;
   s = db_impl_->WriteImpl(write_options_, &rollback_batch, nullptr, nullptr,
-                          no_log_ref, !disable_memtable, &seq_used);
+                          no_log_ref, !DISABLE_MEMTABLE, &seq_used);
   assert(seq_used != kMaxSequenceNumber);
+  if (!s.ok()) {
+    return s;
+  }
   uint64_t& prepare_seq = seq_used;
-  uint64_t commit_seq = GetACommitSeqNumber(prepare_seq);
-  // TODO(myabandeh): skip AddPrepared
-  wpt_db_->AddPrepared(prepare_seq);
-  wpt_db_->AddCommitted(prepare_seq, commit_seq);
+  // Commit the batch by writing an empty batch to the queue that will release
+  // the commit sequence number to readers.
+  WritePreparedCommitEntryPreReleaseCallback update_commit_map(
+      wpt_db_, db_impl_, prepare_seq);
+  WriteBatch empty_batch;
+  empty_batch.PutLogData(Slice());
+  // In the absence of Prepare markers, use Noop as a batch separator
+  WriteBatchInternal::InsertNoop(&empty_batch);
```

Comment:
Why adding a noop? Do you have documentation of the format of WAL especially around the usage of noop?

Reply:
Yeah, here: https://github.com/facebook/rocksdb/blob/master/db/write_batch.cc#L991
Let me know if you have a place in mind that needs to reference this comment.

Reply:
Thanks for the pointer. Since the empty_batch goes by itself, why is the need add a noop? Isn't the write group leader who merge the batches responsible for adding noops?

Reply:
That would an alternative design. The current design is that each writer takes care of the separator. At the time the reasoning was that many batches have natural batch separator with them and there would be no need for extra Noop as separator.